src/sentry/static/sentry/app/views/organizationDashboard/exploreWidget.jsx

```diff
@@ -1,69 +1,119 @@
 import Button from 'app/components/button';
+import DropdownLink from 'app/components/dropdownLink';
 import InlineSvg from 'app/components/inlineSvg';
 import SentryTypes from 'app/sentryTypes';
+import space from 'app/styles/space';
 import withOrganization from 'app/utils/withOrganization';
 
+const exploreMenuCss = css`
+  font-weight: normal;
+  text-transform: none;
+  white-space: nowrap;
+`;
+
 class ExploreWidget extends React.Component {
   static propTypes = {
     widget: SentryTypes.Widget,
+    queries: PropTypes.arrayOf(SentryTypes.DiscoverQuery),
     organization: SentryTypes.Organization,
     selection: SentryTypes.GlobalSelection,
     router: PropTypes.object,
   };
 
-  handleExportToDiscover = event => {
-    const {organization, widget, router} = this.props;
-    const [firstQuery] = widget.queries.discover;
+  getExportToDiscover = query => {
+    const {organization} = this.props;
     const {
       datetime,
       environments, // eslint-disable-line no-unused-vars
       ...selection
     } = this.props.selection;
 
-    event.stopPropagation();
-
     // Discover does not support importing these
     const {
       groupby, // eslint-disable-line no-unused-vars
       rollup, // eslint-disable-line no-unused-vars
       orderby,
-      ...query
-    } = firstQuery;
+      ...restQuery
+    } = query;
 
     const orderbyTimeIndex = orderby.indexOf('time');
     let visual = 'table';
 
     if (orderbyTimeIndex !== -1) {
-      query.orderby = `${orderbyTimeIndex === 0 ? '' : '-'}${query.aggregations[0][2]}`;
+      restQuery.orderby = `${orderbyTimeIndex === 0 ? '' : '-'}${restQuery
+        .aggregations[0][2]}`;
       visual = 'line-by-day';
     } else {
-      query.orderby = orderby;
+      restQuery.orderby = orderby;
     }
 
-    router.push(
-      `/organizations/${organization.slug}/discover/${getQueryStringFromQuery({
-        ...query,
-        ...selection,
-        start: datetime.start,
-        end: datetime.end,
-        range: datetime.period,
-        limit: 1000,
-      })}&visual=${visual}`
-    );
+    return `/organizations/${organization.slug}/discover/${getQueryStringFromQuery({
+      ...restQuery,
+      ...selection,
+      start: datetime.start,
+      end: datetime.end,
+      range: datetime.period,
+      limit: 1000,
+    })}&visual=${visual}`;
+  };
+
+  getExportToEvents = query => {
+    const {selection, organization} = this.props;
+    return getEventsUrlPathFromDiscoverQuery({
+      organization,
+      selection,
+      query,
+    });
   };
 
   render() {
-    // TODO(billy): This is temporary
-    // Need design followups
+    const {widget} = this.props;
+    const discoverQueries = widget.queries.discover;
+
     return (
-      <Button size="xsmall" onClick={this.handleExportToDiscover}>
-        <InlineSvg src="icon-discover" />
-      </Button>
+      <DropdownLink
+        anchorRight={true}
+        title={t('Explore')}
+        topLevelClasses={exploreMenuCss}
```

Comment:
we should make this better one day...

Reply:
This will be changed in the PR to redesign the explore menu.